bundler/lib/dependabot/bundler/native_helpers.rb

```diff
@@ -1,10 +1,47 @@
 # frozen_string_literal: true
 
+require "bundler"
+require "dependabot/shared_helpers"
+
 module Dependabot
   module Bundler
     module NativeHelpers
-      def self.helper_path
-        "bundle exec ruby #{File.join(native_helpers_root, 'run.rb')}"
+      def self.run_bundler_subprocess(function:, args:, bundler_version:)
+        # Run helper suprocess with all bundler-related ENV variables removed
+        with_unbundled_env do
+          SharedHelpers.run_helper_subprocess(
+            command: helper_path(bundler_version: bundler_version),
+            function: function,
+            args: args,
+            env: {
+              # Bundler will pick the matching installed major version
+              "BUNDLER_VERSION" => bundler_version,
+              "BUNDLE_GEMFILE" => File.join(versioned_helper_path(bundler_version: bundler_version), "Gemfile"),
+              "BUNDLE_PATH" => File.join(versioned_helper_path(bundler_version: bundler_version), ".bundle")
```

Comment:
Changed this to match `BUNDLE_GEMFILE` but not sure it's necessary, both seemed to work inside and outside the container with v1 and v2 installed

Reply:
As discussed synchronously, I'm 👍🏻  on having `versioned_helper_path(bundler_version: bundler_version)` as a clean(ish) room bundler path for all things related to the native helper process.